src/authentication.rs

```diff
@@ -35,6 +34,12 @@ static OAUTH_SCOPES: &[&str] = &[
     "user-top-read",
 ];
 
+pub fn get_client_redirect_uri(configuration: &Config) -> Result<String, String> {
```

Comment:
This function always returns `Ok()`, no?
```suggestion
pub fn get_client_redirect_uri(configuration: &Config) -> String {
```

Reply:
yea, fixed